sklearn/cluster/_hdbscan/_tree.pyx

```diff
@@ -251,14 +254,15 @@ cdef dict _compute_stability(
     largest_child = max(largest_child, smallest_cluster)
     births = np.full(largest_child + 1, np.nan, dtype=np.float64)
 
-    births = np.full(largest_child + 1, np.nan, dtype=np.float64)
-    for condensed_node in condensed_tree:
+    for idx in range(PyArray_SHAPE(<cnp.PyArrayObject*> condensed_tree)[0]):
+        condensed_node = condensed_tree[idx]
```

Comment:
Why do you need that ? I'd expect `condensed_tree.shape[0]` to work

Reply:
With the new NumPy API the use of `.shape` errors.

See: https://github.com/scikit-learn/scikit-learn/pull/26385#discussion_r1198437008